cmd/collectors/zapi/plugins/shelf/shelf.go

```diff
@@ -222,13 +223,18 @@ func (my *Shelf) handle7Mode(data *matrix.Matrix, result []*node.Node) ([]*matri
 		data1.Reset()
 	}
 
-	// Purge and reset data
-	my.shelfData.PurgeInstances()
-	my.shelfData.Reset()
+	// reset instance and matrix of shelfMetrics
+	my.shelfMetrics.PurgeInstances()
+	my.shelfMetrics.Reset()
 
-	my.shelfData.SetGlobalLabels(data.GetGlobalLabels())
-	for _, instance := range data.GetInstances() {
-		instance.SetExportable(false)
+	// Purge instances generated from template and updated data metrics from shelfMetrics
+	data.PurgeInstances()
+	for metricName, m := range my.shelfMetrics.GetMetrics() {
+		_, err := data.NewMetricFloat64(metricName, m.GetName())
+		if err != nil {
+			my.Logger.Error().Err(err).Msg("add metric")
```

Comment:
should we continue when there is an error?

Reply:
Mostly we do error out where error occurred while instance creation, as nothing would be possible further that,
But for metrics creation or setting value in plugin, if error occurred then we logged as error and go ahead as it's impacting only that metric.